crates/shared/src/http_solver/model.rs

```diff
@@ -158,7 +158,7 @@ pub struct InteractionData {
     ///
     /// `AMM -> GPv2Settlement`
     pub outputs: Vec<TokenAmount>,
-    pub exec_plan: Option<ExecutionPlan>,
+    pub exec_plan: ExecutionPlan,
```

Comment:
I don't think this needs to be mandatory. Specifically, for `InteractionData`, it shows up in a vector and we its index in a vector for ordering.

Reply:
Hmm, but it is an issue for solutions that mix and match AMM executions and interaction data... So maybe it is needed after all.

Reply:
I would prefer if it is mandatory. It's cleaner, it's obvious how to compute for solvers, and it resolves cases where AMM, 0x orders and interaction data mix and match, as @nlordell said.

Regarding the results.json, I don't have a strong opinion here. I believe Laertes was using driver-provided AMMs at some point, but I guess not anymore. Given though that we have started providing Univ3 liquidity (and maybe solvers consider it as back-up solution), it would be good to announce this change explicitly, if it happens anyway.

Reply:
I would just add (because it's not obvious) that `exec_plan` is made mandatory for non-user orders (0x orders) also:
https://github.com/cowprotocol/services/pull/791/files#diff-fe7a9edd93075fefdfc5a778ca8ec3f906efc33a94be9a1a5513dff787c7261eR214